conf/conf_test.go

```diff
@@ -67,3 +70,294 @@ func TestGetYamlFileFromFile(t *testing.T) {
+      key: /Users/chenhao/.ssh/pem/my.pem
+      threshold:
+        cpu: 0.15
+        mem: 0.70
+        disk: 0.90
+      channels:
+        - general
+        - test
+
+
+client:
+  - name: Redis Native Client (local)
+    driver: "redis"
+    host: "localhost:6379"
+    password: "abc123"
+    channels:
+      - test
+  - name: MySQL Native Client (local)
+    driver: "mysql"
+    host: "localhost:3306"
+    username: "root"
+    password: "pass"
+    ca: /Users/chenhao/Github/mTLS/certs/ca.crt
+    cert: /Users/chenhao/Github/mTLS/certs/client.b.crt
+    key: /Users/chenhao/Github/mTLS/certs/client.b.key
+
+
+notify:
+  aws_sns:
+    - name: AWS SNS
+      region: us-west-2
+      arn: arn:aws:sns:us-west-2:xxx:NotifyMe
+      endpoint: https://sns.us-west-2.amazonaws.com
+      #format: "html"
+      credential:
+        id: xxx
+        key: xxx/yyy
+      dry: true
+  log:
+    - name: "Local Log"
+      file: "/tmp/easeprobe.log"
+      dry: true
+  slack:
+    - name: "MegaEase"
+      webhook: "https://hooks.slack.com/services/xxx"
+      channels:
+        - "slack#coolshell"
+        - "general"
+      #dry: true
+  discord:
+    - name: "MegaEase"
+      webhook: "https://discord.com/api/webhooks/xxx"
+      avatar: "https://img.icons8.com/ios/72/appointment-reminders--v1.png"
+      thumbnail: "https://freeiconshop.com/wp-content/uploads/edd/notification-flat.png"
+      dry: true
+      retry:
+        times: 3
+        interval: 10s
+    - name: "Test"
+      webhook: "https://discord.com/api/webhooks/xxx"
+      avatar: "https://img.icons8.com/ios/72/appointment-reminders--v1.png"
+      thumbnail: "https://freeiconshop.com/wp-content/uploads/edd/notification-flat.png"
+      retry:
+        times: 3
+        interval: 10s
+      channels:
+        - "general"
+  telegram:
+    - name: "Hao Channel"
+      token: 5143540997:AAGv9QXbkFgI0ny6wUG_3IpGP6ci4EhLNs0
+      chat_id: -1001717338903
+      channels:
+        - "telegram#hao"
+        - "test"
+    - name: "Hao Group"
+      token: 5143540997:AAGv9QXbkFgI0ny6wUG_3IpGP6ci4EhLNs0
+      #chat_id: -434882872
+      chat_id: -780609052
+      dry: true
+      retry:
+        times: 3
+        interval: 8s
+  email:
+    - name: "email"
+      server: smtp.email.com:465
+      username: noreply@megaease.com
+      password: xxx
+      to: "test@email.com"
+      dry: true
+
+
+settings:
+  name: "EaseProbeBot"
+  icon: https://upload.wikimedia.org/wikipedia/commons/2/2d/Etcher-icon.png
+  http:
+    ip: 127.0.0.1
+    port: 8181
+    refresh: 5s
+    log:
+      file: /tmp/log/easeprobe.access.log
+      self_rotate: true
+  sla:
+    schedule: "weekly"
+    time: "23:59"
+    debug: true
+    backups: 20
+    channels:
+      - general
+  notify:
+    dry: true
+    retry:
+      times: 5
+      interval: 10s
+  probe:
+    interval: 15s
+  log:
+    level: debug
+    size: 1
+  #backups: 10
+  debug: true
+  timeformat: "2006-01-02 15:04:05 UTC"
+`
```

Comment:
How about we add this configuration into a file so that we can create more configuration files as test cases without changing the code?

Reply:
Actually, I face the dilemma of using a yaml file or just hard coding it.  and considering this is the unit test, we have to code hard the `assert` to check the expectation. for example, I need to check the configuration details. In other words, if we change the yaml file, we must change the code to align with it. So, I put it into the code, it looks a bit ugly, I will think about how to optimize it.

Reply:
I had the same dilemma, when i was looking at this. I leave this up to you :+1: